console/src/permissions/SetLayout.tsx

```diff
@@ -0,0 +1,179 @@
+
+interface SetModalProps extends Partial<Layout.State> {
+  user: user.User;
+}
+
+export const setLayout = ({ user, window, ...rest }: SetModalProps): Layout.State => ({
+  key: SET_LAYOUT_TYPE,
+  type: SET_LAYOUT_TYPE,
+  windowKey: SET_LAYOUT_TYPE,
+  icon: "Access",
+  location: "modal",
+  name: `${user.username}.Permissions`,
+  window: {
+    resizable: false,
+    size: { height: 350, width: 700 },
+    navTop: true,
+    ...window,
+  },
+  args: user,
+  ...rest,
+});
+
+const SAVE_TRIGGER: Triggers.Trigger = ["Control", "Enter"];
+
+const initialPermissions = {
+  schematic: false,
+  admin: false,
+  keys: {},
+};
+
+const formSchema = permissionsZ.extend({
+  keys: consolePolicyKeysZ,
+});
+
+export const SetModal = (props: Layout.RendererProps): ReactElement => {
+  const { layoutKey, onClose } = props;
+  const user_ = Layout.useSelectArgs<user.User>(layoutKey);
+  const addStatus = Status.useAggregator();
+
+  const methods = Form.useSynced<typeof formSchema>({
+    key: [user_.key],
+    name: "Permissions",
+    values: { ...initialPermissions, keys: {} },
+    queryFn: async ({ client }) => {
+      if (client == null) throw new Error("Client is not available");
+      const policies = await client.access.policy.retrieveFor(
+        user.ontologyID(user_.key),
+      );
+      const userSpecificPolicies = policies.filter(
+        (p) => p.subjects.length === 1 && p.subjects[0].key === user_.key,
+      );
+      const keys = convertPoliciesToKeys(userSpecificPolicies);
+      const permissions = convertKeysToPermissions(keys);
+      return {
+        ...permissions,
+        keys,
+      };
+    },
+    applyChanges: async ({ client, values, path, prev }) => {
+      if (path === "") return;
+      const policy = path as ConsolePolicy;
+      const previouslyActive = prev as boolean;
+      if (previouslyActive) {
+        const key = values.keys[policy];
+        if (key == null) return;
+        await client.access.policy.delete(key);
+        return;
+      }
+      const newPolicy = await client.access.policy.create({
+        subjects: {
+          type: user.ONTOLOGY_TYPE,
+          key: user_.key,
+        },
+        ...consolePolicyRecord[policy],
+      });
+      values.keys[policy] = newPolicy.key;
+    },
+  });
+
+  const isRootUser = user_.rootUser;
+  if (isRootUser) {
+    addStatus({
+      variant: "error",
+      message: "Root user permissions cannot be modified",
+    });
+    onClose();
+    return <></>;
+  }
+
+  return (
+    <Align.Space style={{ paddingTop: "2rem", height: "100%" }} grow empty>
+      <Align.Space
+        className="console-form"
+        justify="center"
+        style={{ padding: "3rem" }}
+        grow
+      >
+        <Form.Form {...methods}>
+          <Align.Space direction="y">
+            <Text.Text level="h2">Permissions for {user_.username}</Text.Text>
+            {Array.from(consolePolicySet).map((policy, index) => (
+              <Fragment key={index}>
```

Comment:
The spacing between the permission toggle and the description is strange here. It's also strange that the description draws more visual attention than the permission title.